internal/app/machined/pkg/system/services/apid.go

```diff
@@ -113,7 +113,17 @@ func (o *APID) Runner(config runtime.Configurator) (runner.Runner, error) {
 
 	env := []string{}
 	for key, val := range config.Machine().Env() {
-		env = append(env, fmt.Sprintf("%s=%s", key, val))
+		switch key {
```

Comment:
qq, how could those variables get into environment, as we control the environment when we run the service?

Reply:
ah, got it, it could be user-supplied

Reply:
We support setting environment variables via machine config ( a single set that gets used everywhere ):
Can set them like this:
```
[18:23:34] $ head -20 proxyconfigs/init.yaml 
version: v1alpha1
machine:
  type: init
  token: igdquc.xhmz8aackuvi3x0h
  env:
    http_proxy: http://10.0.0.1:3128
    https_proxy: http://10.0.0.1:3128
```
And then in all of our services we consume `config.Machine().Env()`
```
internal/app/machined/internal/phase/config/extra_env_vars.go:	for key, val := range r.Config().Machine().Env() {
internal/app/machined/pkg/system/services/udevd_trigger.go:	for key, val := range config.Machine().Env() {
internal/app/machined/pkg/system/services/osd.go:	for key, val := range config.Machine().Env() {
internal/app/machined/pkg/system/services/udevd.go:	for key, val := range config.Machine().Env() {
internal/app/machined/pkg/system/services/containerd.go:	for key, val := range config.Machine().Env() {
internal/app/machined/pkg/system/services/ntpd.go:	for key, val := range config.Machine().Env() {
internal/app/machined/pkg/system/services/etcd.go:	for key, val := range config.Machine().Env() {
internal/app/machined/pkg/system/services/trustd.go:	for key, val := range config.Machine().Env() {
internal/app/machined/pkg/system/services/kubelet.go:	for key, val := range config.Machine().Env() {
internal/app/machined/pkg/system/services/system-containerd.go:	for key, val := range config.Machine().Env() {
internal/app/machined/pkg/system/services/apid.go:	for key, val := range config.Machine().Env() {
internal/app/machined/pkg/system/services/networkd.go:	for key, val := range config.Machine().Env() {
```

Reply:
Since APId is the only service that connects to other gRPC endpoints, it's the only one we need to filter atm.